apps/nextjs-website/src/lib/strapi/seo.ts

```diff
@@ -0,0 +1,64 @@
+import * as t from 'io-ts/lib';
+import { NullToUndefinedCodec } from './codecs/NullToUndefinedCodec';
+
+export const SEOImageFormatCodec = t.partial({
+  name: t.union([NullToUndefinedCodec, t.string]),
+  hash: t.union([NullToUndefinedCodec, t.string]),
+  ext: t.union([NullToUndefinedCodec, t.string]),
+  mime: t.union([NullToUndefinedCodec, t.string]),
+  width: t.union([NullToUndefinedCodec, t.number]),
+  height: t.union([NullToUndefinedCodec, t.number]),
+  size: t.union([NullToUndefinedCodec, t.number]),
+  path: t.union([NullToUndefinedCodec, t.string]),
+  url: t.union([NullToUndefinedCodec, t.string]),
+});
+
+export const SEOImageCodec = t.partial({
+  id: t.union([NullToUndefinedCodec, t.number]),
+  name: t.union([NullToUndefinedCodec, t.string]),
+  alternativeText: t.union([NullToUndefinedCodec, t.string]),
+  caption: t.union([NullToUndefinedCodec, t.string]),
+  width: t.union([NullToUndefinedCodec, t.number]),
+  height: t.union([NullToUndefinedCodec, t.number]),
+  hash: t.union([NullToUndefinedCodec, t.string]),
+  ext: t.union([NullToUndefinedCodec, t.string]),
+  mime: t.union([NullToUndefinedCodec, t.string]),
+  size: t.union([NullToUndefinedCodec, t.number]),
+  url: t.union([NullToUndefinedCodec, t.string]),
+  previewUrl: t.union([NullToUndefinedCodec, t.string]),
+  provider: t.union([NullToUndefinedCodec, t.string]),
+  provider_metadata: t.union([NullToUndefinedCodec, t.unknown]),
+  formats: t.partial({
+    thumbnail: SEOImageFormatCodec,
+    small: SEOImageFormatCodec,
+    medium: SEOImageFormatCodec,
+  }),
+});
```

Comment:
I think you should search for differences with MediaAttributeCodec and use it instead of SEOImageCodec.
If you find some differences you should use MediaAttributesCodec as a base codec and add more attributes using an intersection

Reply:
You should also take a look at `apps/nextjs-website/src/lib/strapi/codecs/BlocksContentCodec.ts` and if possible, do a refactor by replacing `ImageBlockNodeCodec` with `MediaAttributeCodec`:

```export const ImageBlockNodeCodec = t.strict({
  type: t.literal('image'),
  image: t.intersection([
    t.strict({
      name: t.string,
      url: t.string,
      width: t.number,
      height: t.number,
      hash: t.string,
      ext: t.string,
      mime: t.string,
      size: t.number,
      provider: t.string,
      createdAt: t.string,
      updatedAt: t.string,
      caption: t.union([NullToUndefinedCodec, t.string]),
      previewUrl: t.union([NullToUndefinedCodec, t.string]),
    }),
    t.partial({
      alternativeText: t.string,
      formats: t.record(t.string, t.unknown),
      provider_metadata: t.unknown,
    }),
  ]),
  children: t.array(ImageBlockChildCodec),
});
```

Reply:
Is this refactor possible?